### PR TITLE
table/tables: migrate test-infra to testify for partition_test.go

### LINKED ISSUE #27589: migrate test-infra to testify for `table/tables/partition_test.go `

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #27589 

based on #27557

### Release note

```release-note
None
```

### CODE REVIEW

File: testkit/testkit.go [hunk 1]
```diff
@@ -176,3 +177,14 @@ func newSession(t *testing.T, store kv.Storage) session.Session {
 	se.SetConnectionID(testKitIDGenerator.Inc())
 	return se
 }
+
+// MustGetErrCode executes a sql statement and assert it's error code.
+func (tk *TestKit) MustGetErrCode(sql string, errCode int) {
```
Comment (tisonkun) [inline testkit/testkit.go:182]: This function doesn't used?

Comment (feitian124) [reply testkit/testkit.go:182]: used in `table/tables/partition_test.go` line 509

**Comment by feitian124** [comment]
`check_dev_2` failed for long tests, but should not cased by this change.

```shell

[2021-09-07T15:17:12.872Z] grep 'PASS:' gotest.log | go run tools/check/check-timeout.go || { $(find $PWD/ -type d | grep -vE "(\.git|tools)" | xargs tools/bin/failpoint-ctl disable); exit 1; }

[2021-09-07T15:17:12.872Z] The following test cases take too long to finish:

[2021-09-07T15:17:12.872Z] --- PASS: TestSingle (6.05s)

[2021-09-07T15:17:12.872Z] --- PASS: TestApply (5.08s)

[2021-09-07T15:17:12.872Z] --- PASS: TestPKIsHandleRangeScan (5.09s)

[2021-09-07T15:17:12.872Z] --- PASS: TestSort (5.08s)
```

**Review by tisonkun — APPROVED** [review]
Thanks for your contribution @feitian124 ! LGTM.

**Review by mmyj — APPROVED** [review]
lgtm

**Pull request merged**